sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -348,9 +340,8 @@ class RocksDB(
    * Drop uncommitted changes, and roll back to previous version.
    */
   def rollback(): Unit = {
-    closePrefixScanIterators()
-    resetWriteBatch()
     numKeysOnWritingVersion = numKeysOnLoadedVersion
+    loadedVersion = -1L
```

Comment:
Does it mean we will load a checkpoint from DFS, right?

Reply:
Yes correct

Reply:
Yeah since we no longer use WriteBatch, we are unable to roll back any writes and have to load from DFS. We think the rate of failure is small enough to tolerate the cost, unless the query (or infrastructure) is unstable and cannot make progress in any way.

Reply:
Does any local files need to be cleaned up if they've been written to disk during the task?

Reply:
I don't think we need to do anything explicitly here. Basically the call to load the rocksdb version from dfs should ensure that the appropriate cleanup is done.

```
    val metadata = if (version == 0) {
      if (localDir.exists) Utils.deleteRecursively(localDir)
      localDir.mkdirs()
      RocksDBCheckpointMetadata(Seq.empty, 0)
    } else {
      // Delete all non-immutable files in local dir, and unzip new ones from DFS commit file
      listRocksDBFiles(localDir)._2.foreach(_.delete())
```

If the instance is removed, we would eventually call `close` which would delete the base root dir itself.

cc - @HeartSaVioR - in case there is any condition I may be missing here

Reply:
I guess we are just reusing the mechanism of re-load, which has been existing for a while. If anyone figures out any possibility of bug then it'd be awesome to have a fix, but it sounds to me as out of scope even if there is a bug.